pkg/resource/resource.go

```diff
@@ -60,9 +60,9 @@ const (
 // doesn't matter whether the namespace belongs to a user or organization. This
 // refactor should be completed by August 2024.
 type Namespace struct {
```

Comment:
Not sure if it will be better to make the system variables flatten rather than nested.

Reply:
Because of this change, we also have to changed the pipeline-backend-cloud

Reply:
If there are no blockers for the nested structure, I prefer it because it groups fields that are related (otherwise we duplicate `vars.PipelineRequesterType`, `vars.PipelineRequesterUID`, etc.).

Adding the JSON tags here adds some coupling but I'm fine with it. But perhaps we should follow the convention in `pkg/recipe` and have the tags in UPPER_SNAKE_CASE